src/serenity/config.hpp

```diff
@@ -37,14 +36,32 @@ class SerenityConfig {
    * Overlapping custom configuration options using recursive copy.
    */
   void applyConfig(const SerenityConfig& customCfg) {
-    this->recursiveCfgCopy(this, customCfg);
+    recursiveCfgCopy(this, customCfg);
   }
 
   /**
-   * Gets variant config value.
+   * Templated, safe getter for item in config.
    */
-  Option<SerenityConfig::CfgVariant> operator()(std::string key) const {
-    return getField(key);
+  template <typename T>
+  const Result<T> item(std::string key) const {
```

Comment:
Or even better: overload ```Result<t> operator[] (const std::string&);``` and add ```getSection(const std::string);```, because sections are going to be much much less used.